cmd/mirror-main.go

```diff
@@ -743,7 +743,7 @@ func getEventPathURLWin(srcURL, eventPath string) string {
 }
 
 // runMirror - mirrors all buckets to another S3 server
-func runMirror(srcURL, dstURL string, cli *cli.Context, encKeyDB map[string][]prefixSSEPair) bool {
+func runMirror(ctx context.Context, cancelMirror context.CancelFunc, srcURL, dstURL string, cli *cli.Context, encKeyDB map[string][]prefixSSEPair) bool {
```

Comment:
Not really a fan of sending a `context.CancelFunc` What is the reason for that?

The typical assumption around contexts are that they are immutable when passed down, meaning that `runMirror` shouldn't be modifying the sent context. I assume this is what the cancelfunc does.

If you want to signal a cancelation upstream you typically so that by returning an error. Is that feasible?

Reply:
@klauspost, 

Defining `ctx` and `cancel<mc-command>` in `main<mc-command>` function is the standard way for almost all mc commands.

For `mc mirror` command though, `ctx` and `cancelMirror` were defined in 2 files twice in `mainMirror()` and in `runMirror()` functions. I noticed the duplicate definition and fixed it by removing the 2nd one in `runMirror` function. However, removing 2nd definition caused a compile error and needed to be fixed by passing it through `runMirror().

It is totally an unrelated change and not needed to fix issue #3237. I can remove it if it is desired, since it is an independent problem.

Reply:
It is still a weird pattern, but I guess it works.